aidial_sdk/application.py

```diff
@@ -68,6 +75,49 @@ def add_chat_completion(
     ) -> None:
         self.chat_completion_impls[deployment_name] = impl
 
+    async def _rate_response(
+        self, deployment_id: str, original_request: Request
+    ) -> Response:
+        set_log_deployment(deployment_id)
+        impl = self.chat_completion_impls.get(deployment_id, None)
+
+        if not impl:
+            return JSONResponse(
```

Comment:
These three error handlers _(deployment not found, json encoding error and pydantic validation error)_ are now repeated twice in the the same module.

Let's decompose them into dedicated helpers, like `self._get_deployment`, `DIALApp._get_request_json(request)` and `DIALApp._parse_request(type, request_json)`.

Reply:
fixed